### DISCUSSION TITLE
Neural Network diagrams

### DISCUSSION THREAD

**Discussion by nikhilmitrax** [terrastruct/d2#163] [Feature Requests]
D2 is a fantastic tool, and has a real potential to become the defacto standard for building and sharing diagrams.

To that effect, I was just curious if there were plans to support neural network style diagrams. As far as I understand from looking at the documentation, the current exposed configurations/CSS would not be enough to make professional quality NN diagrams.

For a few examples of what kind of digrams I'd like to make, check out http://alexlenail.me/NN-SVG/index.html

**Comment by alixander** [comment]
i've taken a look at some of these, and my impression is that it's just... layers of circles? if you had every customization on that site possible in D2, would that be feature-complete for neural net diagrams? if so, this should ~0.5 day work.

(also, ty for the kind words)

**Reply by nikhilmitrax** [reply]
I believe the FCNN style of NN diagrams would be fairly trivial to do as you said, but I'm presuming the LeNet and AlexNet style might be a bit more involved?

Regarding the feature complete part, I'm sure there could be a lot more done, but I do believe what's on the page (FCNN style + AlexNet Style + LeNet style) is a big chunk of the feature-set a typical data scientist would require.

**Reply by alixander** [reply]
ah didn't toggle to those. the AlexNet looks like a 3D specific thing. will take some deeper look for me to understand these diagrams. which ones do you personally want to use?

**Reply by nikhilmitrax** [reply]
For starters, FCNN and LeNet are probably the most useful, in that order.
